src/lib.rs

```diff
@@ -101,8 +101,8 @@ where
             block!({
                 // Some implementations (stm32f0xx-hal) want a matching read
                 // We don't want to block so we just hope it's ok this way
-                self.spi.read().ok();
-                self.spi.send(patterns[bits as usize])
+                self.spi.send(patterns[bits as usize]);
```

Comment:
Oh, this won't work properly.

`block!` will call `send` & `read` until `read` doesn't block any more, which means the same byte might be sent multiple times.

Reply:
All right, I'll be honest and say I don't understand any of the inner workings (not yet anyway). It just happened to work for my use case then ;) 
..whereas the original code did not. I can try some more poking around, unless you have any hints/suggestions?


Btw, I've only found one other guy on the internet briefly describing the same issue as I'm having (before he gives up and uses the timer variant): https://albertskog.se/embedded-rust-timer-timeout-problem/